internal/temporalite/lite_server.go

```diff
@@ -0,0 +1,395 @@
+	var namespaces []*sqlite.NamespaceConfig
+	for _, ns := range liteConfig.Namespaces {
+		namespaces = append(namespaces, sqlite.NewNamespaceConfig(liteConfig.BaseConfig.ClusterMetadata.CurrentClusterName, ns, false))
+	}
+	if err := sqlite.CreateNamespaces(sqlConfig, namespaces...); err != nil {
+		return nil, fmt.Errorf("error creating namespaces: %w", err)
+	}
+
+	authorizer, err := authorization.GetAuthorizerFromConfig(&liteConfig.BaseConfig.Global.Authorization)
+	if err != nil {
+		return nil, fmt.Errorf("unable to instantiate authorizer: %w", err)
+	}
+
+	claimMapper, err := authorization.GetClaimMapperFromConfig(&liteConfig.BaseConfig.Global.Authorization, liteConfig.Logger)
+	if err != nil {
+		return nil, fmt.Errorf("unable to instantiate claim mapper: %w", err)
+	}
+
+	serverOpts := []temporal.ServerOption{
+		temporal.WithConfig(liteConfig.BaseConfig),
+		temporal.ForServices(temporal.DefaultServices),
+		temporal.WithLogger(liteConfig.Logger),
+		temporal.WithAuthorizer(authorizer),
+		temporal.WithClaimMapper(func(cfg *config.Config) authorization.ClaimMapper {
+			return claimMapper
+		}),
+	}
+
+	if len(liteConfig.DynamicConfig) > 0 {
+		// To prevent having to code fall-through semantics right now, we currently
+		// eagerly fail if dynamic config is being configured in two ways
+		if liteConfig.BaseConfig.DynamicConfigClient != nil {
+			return nil, fmt.Errorf("unable to have file-based dynamic config and individual dynamic config values")
+		}
+		serverOpts = append(serverOpts, temporal.WithDynamicConfigClient(liteConfig.DynamicConfig))
+	}
+
+	if len(opts) > 0 {
+		serverOpts = append(serverOpts, opts...)
+	}
+
+	srv, err := temporal.NewServer(serverOpts...)
+	if err != nil {
+		return nil, fmt.Errorf("unable to instantiate server: %w", err)
+	}
+
+	s := &LiteServer{
+		internal:         srv,
+		frontendHostPort: liteConfig.BaseConfig.PublicClient.HostPort,
+	}
+
+	return s, nil
+}
+
+// Start temporal server.
+func (s *LiteServer) Start() error {
+	// We wrap Server instead of simply embedding it in the LiteServer struct so
+	// that it's possible to add additional lifecycle hooks here if necessary.
+	return s.internal.Start()
+}
+
+// Stop the server.
+func (s *LiteServer) Stop() {
+	// We wrap Server instead of simply embedding it in the LiteServer struct so
+	// that it's possible to add additional lifecycle hooks here if necessary.
+	s.internal.Stop()
+}
+
+// NewClient initializes a client ready to communicate with the Temporal
+// server in the target namespace.
+func (s *LiteServer) NewClient(ctx context.Context, namespace string) (client.Client, error) {
+	return s.NewClientWithOptions(ctx, client.Options{Namespace: namespace})
+}
+
+// NewClientWithOptions is the same as NewClient but allows further customization.
+//
+// To set the client's namespace, use the corresponding field in client.Options.
+//
+// Note that the HostPort and ConnectionOptions fields of client.Options will always be overridden.
```

Comment:
It's not; I wanted to leave these open to be overridden in the future (eg. if we wanted to add a higher level option for enabling TLS). But actually I think we can do that without a breaking change since TLS config would be introduced along with a new server option, so I'll just update the godoc comment.